Tasks/Common/AzureRmDeploy-common/webdeployment-common/utility.ts

```diff
@@ -182,6 +183,13 @@ export async function generateTemporaryFolderForDeployment(isFolderBasedDeployme
     if(isFolderBasedDeployment || packageType === PackageType.jar) {
         tl.debug('Copying Web Packge: ' + webDeployPkg + ' to temporary location: ' + folderPath);
         copyDirectory(webDeployPkg, folderPath);
+        if(packageType === PackageType.jar) {
+            let src = path.join(folderPath, getFileNameFromPath(webDeployPkg));
+            let dest = path.join(folderPath, "app.jar")
```

Comment:
I see we are taking the approach of renaming the jar file to app.jar. Does it impact the application. Typically the app name acts as virtual path (ex: webpp.azurewebsites.net/<appName>). With renaming it to default app.jar, I presume there wouldn't be no virtual path once the app deploys. Is my understanding right?

Reply:
As per my understanding, package name cannot act as virtual path, since package is not unique. This is one of the recommendations by app service team. Also azure maven plugin team is doing the same.